### PR TITLE
ft(users.tsx):Add role based access control functionality(RBAC)-[187364818]

### PR DESCRIPTION
**Pull request by tuyishimehope**
## ℹ️ Description

Please write about the changes implemented in this PR. Also write a short description about why the change is implemented.

##### 📹 Loom video link: [_URL here_
](https://www.loom.com/share/8c5eb617e1504c5fa066717bac80ebe9)

## 🧪 How can this be tested?

Write a short description on how others can test the changes implemented in the PR

## 📸 Screenshots

image
image
image

## 📍 Type of change

Please check the options that are relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## ✅ Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have created a loom video for the new feature or enhancement
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/layouts/AdminLayout.tsx
Comment (Oliviier-dev) [inline src/layouts/AdminLayout.tsx:1]: Restrict access to admin pages to ensure only admins can view them

**Review by Oliviier-dev — COMMENTED** [review]
Good job far, work on the tests as well as the review down below

**Review by shebz2023 — DISMISSED** [review]
LGTM

**Review by Oliviier-dev — DISMISSED** [review]
LGTM

File: src/components/admincomponents/Navbar.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+              height="32"
+              className="rounded-full ring ring-slate-800"
+              alt="Avatar"
+            />
+            <span className="sr-only">Toggle user menu</span>
+          </button>
+        </div>
+        <button onClick={onClick} className="lg:hidden">
+          <IoMenuOutline size={24} />
+        </button>
```
Comment (SergeInGithub) [inline src/components/admincomponents/Navbar.tsx]: We have a button component

File: src/components/admincomponents/Sidebar.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                className="flex items-center"
+                onClick={handleLogout}
+                disabled={isLoading || logoutLoading}
+                data-testid="logout-button"
+              >
+                <IoLogOutOutline className="text-2xl" />
+                <span className="ml-2">
+                  {isLoading ? 'Logging out...' : 'Logout'}
+                </span>
+              </button>
```
Comment (SergeInGithub) [inline src/components/admincomponents/Sidebar.tsx]: WE have a button component

File: src/components/admincomponents/TableComponents/ActionButton.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+          />
+        </div>
+      )}
+      <button
+        onClick={handleAction}
+        className="p-2 rounded-full hover:bg-gray-200 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
+        role="action"
+      >
+        <BiDotsHorizontalRounded className="h-6 w-6" />
+      </button>
```
Comment (SergeInGithub) [inline src/components/admincomponents/TableComponents/ActionButton.tsx]: Button component instead

File: src/components/admincomponents/TableComponents/ActionButton.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+      {dropdownActive && (
+        <div className="absolute right-0 mb-24 mr-8 w-48 bg-white border border-gray-200 rounded-md shadow-lg z-10">
+          <div className="py-1">
+            <button
+              onClick={handleAssignRoles}
+              className="w-full text-left px-4 py-2 text-sm text-gray-700 hover:bg-gray-100"
+            >
+              Assign Roles
+            </button>
```
Comment (SergeInGithub) [inline src/components/admincomponents/TableComponents/ActionButton.tsx]: Button component instead

File: src/components/admincomponents/TableComponents/RoleModal.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            onClick={() => onAssignRole('USER')}
+            disabled={current === 'USER'}
+            className={`w-full px-4 py-2 text-sm rounded ${
+              current === 'USER'
+                ? 'bg-gray-300 text-gray-700 cursor-not-allowed'
+                : 'bg-blue-500 text-white hover:bg-blue-600'
+            }`}
+          >
+            User
+          </button>
```
Comment (SergeInGithub) [inline src/components/admincomponents/TableComponents/RoleModal.tsx]: Here as well

File: src/components/admincomponents/TableComponents/RoleModal.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  current,
+}) => {
+  if (!isOpen) return null;
+
+  return (
+    <div className="fixed inset-0 flex items-center justify-center z-50">
+      <div
+        className="bg-black bg-opacity-50 absolute inset-0"
+        onClick={onClose}
+      ></div>
```
Comment (SergeInGithub) [inline src/components/admincomponents/TableComponents/RoleModal.tsx]:
```suggestion
      <div
        className="bg-black bg-opacity-50 absolute inset-0"
        onClick={onClose}
      />
```

File: src/components/admincomponents/TableComponents/RoleModal.tsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+            onClick={() => onAssignRole('VENDOR')}
+            disabled={current === 'VENDOR'}
+            className={`w-full px-4 py-2 text-sm rounded ${
+              current === 'VENDOR'
+                ? 'bg-gray-300 text-gray-700 cursor-not-allowed'
+                : 'bg-yellow-500 text-white hover:bg-yellow-600'
+            }`}
+          >
+            Vendor
+          </button>
```
Comment (SergeInGithub) [inline src/components/admincomponents/TableComponents/RoleModal.tsx]: Button component instead

File: src/components/admincomponents/TableComponents/RoleModal.tsx [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        </div>
+        <button
+          onClick={onClose}
+          className="mt-4 w-full px-4 py-2 text-sm bg-gray-500 text-white rounded hover:bg-gray-600"
+        >
+          Close
+        </button>
```
Comment (SergeInGithub) [inline src/components/admincomponents/TableComponents/RoleModal.tsx]: Here as well

File: src/layouts/AdminLayout.tsx
Comment (SergeInGithub) [reply src/layouts/AdminLayout.tsx:1]: Work on this as well

**Review by SergeInGithub — CHANGES_REQUESTED** [review]
Nice work @tuyishimekyrie Just work on some minor reviews below and you're good to go

File: src/components/admincomponents/TableComponents/RoleModal.tsx (hunk 6 above)
Comment (SergeInGithub) [reply src/components/admincomponents/TableComponents/RoleModal.tsx]: Can you work on this please!

**Review by SergeInGithub — CHANGES_REQUESTED** [review]
Please finish all reviews! @tuyishimekyrie

**Review by SergeInGithub — APPROVED** [review]
LGTM

**Pull request merged**